test/api/controllers/adminController.test.js

```diff
@@ -833,15 +833,18 @@ describe('Test: admin controller', () => {
       return adminController.updateSpecifications(requestObject)
         .catch(error => {
           try {
-            should(kuzzle.pluginsManager.trigger).be.calledOnce();
+            should(kuzzle.pluginsManager.trigger).be.calledTwice();
             should(kuzzle.pluginsManager.trigger.firstCall).be.calledWith('data:beforeUpdateSpecifications', requestObject);
+            should(kuzzle.pluginsManager.trigger.secondCall).be.calledWith('validation:error', 'Some errors with provided specifications.');
             should(kuzzle.internalEngine.refresh).not.be.called();
             should(kuzzle.validation.curateSpecification).not.be.called();
             should(kuzzle.internalEngine.createOrReplace).not.be.called();
 
-            should(error).be.an.instanceOf(BadRequestError);
-            should(error.message).be.exactly('Some errors with provided specifications.');
-            should(error.details).match([ 'bad bad is a bad type !' ]);
+            // we do not want a BadRequest Error here: the request is good, the content is not
+            should(error).be.an.instanceOf(ResponseObject);
```

Comment:
I disagree. Promises should be rejected only with error objects. Rejecting with a `ResponseObject` seems counterintuitive to me, making the code harder to understand and to maintain.